bdk_esplora_example/src/esplora.rs

```diff
@@ -163,13 +161,19 @@ impl Client {
                             },
                             false => ConfirmationTime::Unconfirmed,
                         };
-                        if let Err(err) = update.insert_tx(tx.to_tx(), Some(confirmation_time)) {
-                            match err {
-                                InsertTxErr::TxTooHigh => {
-                                    /* Don't care about new transactions confirmed while syncing */
+                        if let Err(failure) = update.insert_tx(tx.to_tx(), confirmation_time) {
+                            use bdk_chain::{
+                                chain_graph::InsertTxFailure, sparse_chain::InsertTxFailure::*,
+                            };
+                            match failure {
+                                InsertTxFailure::Chain(TxTooHigh { .. }) => {
+                                    /* Chain tip has increased, ignore tx for now */
+                                }
+                                InsertTxFailure::Chain(TxMovedUnexpectedly { .. }) => {
+                                    /* Reorg occured (catch error below), ignore tx for now */
                                 }
-                                InsertTxErr::TxMoved => {
-                                    /* This means there is a reorg, we will catch that below */
+                                InsertTxFailure::UnresolvableConflict(_) => {
+                                    /* Reorg occured (catch error below), ignore tx for now */
```

Comment:
I'm not 100% certain this implies there has been a re-org. It just means there is a confirmed transaction that conflicts with it.

Reply:
I'm assuming it would either be a reorg, a bug, or a malicious node/server we are connected to, correct? 😅

Reply:
I think essplora will actually return stuff that has recently been evicted. See #128